samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -0,0 +1,515 @@
+{
+#if defined(CONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS)
+	*num_tx_pkts = CONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS;
+	if (*num_tx_pkts == 0) {
+		LOG_ERR("Can't send %d number of raw tx packets", *num_tx_pkts);
+		return -1;
+	}
+	LOG_INF("Sending %d number of raw tx packets", *num_tx_pkts);
+#else
+	*num_tx_pkts = -1;
+	LOG_INF("Sending raw tx packets continuously");
+#endif
+	return 0;
+}
+
+static void wifi_send_raw_tx_packet(void)
+{
+	struct sockaddr_ll sa;
+	int sockfd, ret;
+	struct nrf_wifi_fmac_rawpkt_info packet;
+	char *test_frame = NULL;
+	int buf_length, num_pkts, transmission_mode;
+
+	ret = setup_raw_pkt_socket(&sockfd, &sa);
+	if (ret < 0) {
+		LOG_ERR("Setting socket for raw pkt transmission failed %d", errno);
+		return;
+	}
+
+	fill_raw_tx_pkt_hdr(&packet);
+
+	ret = print_pkt_count_info(&num_pkts);
+	if (ret < 0) {
+		close(sockfd);
+		return;
+	}
+
+	test_frame = malloc(sizeof(struct nrf_wifi_fmac_rawpkt_info) + sizeof(test_beacon_frame));
+	if (!test_frame) {
+		LOG_ERR("Malloc failed for send buffer %d", errno);
+		return;
+	}
+
+	buf_length = sizeof(struct nrf_wifi_fmac_rawpkt_info) + sizeof(test_beacon_frame);
+	memcpy(test_frame, &packet, sizeof(struct nrf_wifi_fmac_rawpkt_info));
+	memcpy(test_frame + sizeof(struct nrf_wifi_fmac_rawpkt_info),
+				&test_beacon_frame, sizeof(test_beacon_frame));
+
+	transmission_mode = CONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE;
+	while (1) {
+		if (transmission_mode == FIXED_MODE_TRANSMISSION && num_pkts == 0) {
+			break;
+		}
+
+		ret = sendto(sockfd, test_frame, buf_length, 0,
+				(struct sockaddr *)&sa, sizeof(sa));
+		if (ret < 0) {
+			LOG_ERR("Unable to send beacon frame: %s", strerror(errno));
+			close(sockfd);
+			free(test_frame);
+			return;
+		}
+
+		if (transmission_mode == FIXED_MODE_TRANSMISSION && num_pkts > 0) {
+			num_pkts--;
+		}
+
+		k_msleep(CONFIG_RAW_TX_PACKET_SAMPLE_INTER_FRAME_DELAY_MS);
+	}
+
+	/* close the socket */
+	close(sockfd);
+	free(test_frame);
+}
+
+int main(void)
+{
+	memset(&context, 0, sizeof(context));
+
+	net_mgmt_init_event_callback(&wifi_shell_mgmt_cb,
+				     wifi_mgmt_event_handler,
+				     WIFI_SHELL_MGMT_EVENTS);
+
+	net_mgmt_add_event_callback(&wifi_shell_mgmt_cb);
+
+	net_mgmt_init_event_callback(&net_shell_mgmt_cb,
+				     net_mgmt_event_handler,
+				     NET_EVENT_IPV4_DHCP_BOUND);
+
+	net_mgmt_add_event_callback(&net_shell_mgmt_cb);
+
+	LOG_INF("Starting %s with CPU frequency: %d MHz", CONFIG_BOARD, SystemCoreClock/MHZ(1));
+	k_sleep(K_SECONDS(1));
+
+	LOG_INF("Static IP address (overridable): %s/%s -> %s",
+		CONFIG_NET_CONFIG_MY_IPV4_ADDR,
+		CONFIG_NET_CONFIG_MY_IPV4_NETMASK,
+		CONFIG_NET_CONFIG_MY_IPV4_GW);
+
+	wifi_set_mode();
+
+#ifdef CONFIG_RAW_TX_PACKET_SAMPLE_CONNECTION_MODE
+	while (1) {
+		wifi_connect();
+
+		while (!context.connect_result) {
+			cmd_wifi_status();
+			k_sleep(K_MSEC(STATUS_POLLING_MS));
+		}
+
+		if (context.connected) {
+			wifi_send_raw_tx_packet();
```

Comment:
I know we are following this model in all samples, but this makes it tough to read, I propose below which separates the actual operation from connection.

```
try_wifi_connect()
{
  while (1) {
      ....
      ....
  }
}

status = try_wifi_connect();
if (status != SUCCESS) 
    return -1;

wifi_send_raw_tx_packet();
```